src/keys.rs

```diff
@@ -332,25 +335,26 @@ impl IssuanceValidatingKey {
     pub fn to_bytes(&self) -> [u8; 32] {
         // This is correct because the wrapped point must have ỹ = 0, and
         // so the point repr is the same as I2LEOSP of its x-coordinate.
-        <[u8; 32]>::from(&self.0)
+        <U256 as FieldBytesEncoding<Secp256k1>>::decode_field_bytes(&self.0.to_bytes())
+            .to_be_bytes()
```

Comment:
Look too complicated
Could you check whether `self.0.to_bytes().into()` returns the bytes in big endian order